# workspace: [-1.0, 1.0, -0.6, 0.6, 0.0, 0.77]

# # measured parameters
# gp_os: 0.137+0.02175
# gp_tip_w: 0.01
# da_os_x: 0.0035
# da_os_z: 0.009
# target_to_edge: 0.014
# target_diameter: 0.01
# eyestay_thickness: 0.01
# eyestay_opening: 0.05
# table_offset: 0.025 # -0.01
# scene_centre: [0.5, 0.0, 0.2]
# sl_length: 1.34 # = 1.42-0.08. 0.06 for 2 heads and 0.02 as safe boundary
# aglet_thickness: 0.0045
# aglet_length: 0.03

# # primitive parameters  
# app_os: 0.05 # approach offset
# grasp_rot_l: [0, pi, pi/2]
# grasp_rot_r: [0, pi, -pi/2]
# insert_pitch2: pi/24*10 # high top pi/24*12, low top pi/24*10

# # predefined poses
# grasp_states: [-0.6677426099777222, -1.143572211265564, 1.6579793691635132, 0.6495829820632935,
#   2.425766944885254, 0.5661510825157166, 2.1443932056427, 0.6723159551620483, -1.0775535106658936,
#   -1.690792202949524, 0.6427130103111267, 3.8197340965270996, 0.521094024181366, -2.0585293769836426]
# observe_states: [-0.720012903213501, -0.3579239249229431, 1.772327184677124, 0.13136780261993408, 
#   -0.648350715637207, 1.1232486963272095, -1.8344221115112305, 0.8814099431037903, -0.36250463128089905, 
#   -1.85231351852417, -0.028060272336006165, 0.606028139591217, 1.1636393070220947, 1.9661468267440796]
# grasp_states2: [-1.0771714448928833, -0.7815413475036621, 1.3724327087402344, 0.3715556263923645, 
#   1.7982889413833618, 0.8510035872459412, 2.2453041076660156, 1.0602233409881592, -0.6867497563362122, 
#   -1.4237779378890991, 0.3548184633255005, 4.543345928192139, 0.7604496479034424, -2.2329251766204834]

# ws[1] - 
# ws[2] - 
workspace: [-1.0, 1.0, -0.6, 0.6, 0.0, 0.77] # workspace limits [xmin, xmax, ymin, ymax, zmin, zmax]
gp_os: 0.137+0.02175
gp_tip_w: 0.01
da_os_x: 0.0035
da_os_z: 0.009


scene_centre: [0.5, 0.0, 0.35]
scene_centre2: [0.46336247, -0.01777989, 0.01] # for the second scene

rope_length: 0.315

target_holder_thickness: 0.014
target_to_holder: 0.026
target_diameter: 0.0035
target_thickness: 0.015
table_offset: 0.025
marker_length: 0.025
marker_thickness: 0.007


# primitive parameters
app_os: 0.05 # approach offset
grasp_rot_l: [0, pi, pi/2]
grasp_rot_r: [0, pi, -pi/2]
insert_pitch2: pi/24*10 # high top pi/24*12, low top pi/24*10

# predefined poses
grasp_states: [-0.6677426099777222, -1.143572211265564, 1.6579793691635132, 0.6495829820632935,
  2.425766944885254, 0.5661510825157166, 2.1443932056427, 0.6723159551620483, -1.0775535106658936,
  -1.690792202949524, 0.6427130103111267, 3.8197340965270996, 0.521094024181366, -2.0585293769836426]
observe_states: [-0.720012903213501, -0.3579239249229431, 1.772327184677124, 0.13136780261993408, 
  -0.648350715637207, 1.1232486963272095, -1.8344221115112305, 0.8814099431037903, -0.36250463128089905, 
  -1.85231351852417, -0.028060272336006165, 0.606028139591217, 1.1636393070220947, 1.9661468267440796]
grasp_states2: [-1.0771714448928833, -0.7815413475036621, 1.3724327087402344, 0.3715556263923645, 
  1.7982889413833618, 0.8510035872459412, 2.2453041076660156, 1.0602233409881592, -0.6867497563362122, 
  -1.4237779378890991, 0.3548184633255005, 4.543345928192139, 0.7604496479034424, -2.2329251766204834]